projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -66,35 +69,45 @@ export class IgxOverlayService {
     /**
      * Shows the provided component.
      */
-    show(component: ElementRef | Type<{}>, settings?: OverlaySettings): string {
-        const id: string = (this._componentId++).toString();
-        settings = Object.assign({}, this._defaultSettings, settings);
+    show(component: ElementRef | Type<{}>, settings?: OverlaySettings, id?: string): string {
+        let info: OverlayInfo;
+        if (id) {
```

Comment:
If we pass 0 as Id this will evaluate as false - is that expected? If not use if (id !== undefined)

Reply:
fixed